Sources/SKCore/BuildSystem.swift

```diff
@@ -35,5 +35,14 @@ public protocol BuildSystem {
   /// Returns the toolchain to use to compile this file
   func toolchain(for: URL, _ language: Language) -> Toolchain?
 
-  // TODO: notifications when settings change.
+  /// Delegate to handle any build system events such as file build settings changing.
+  var delegate: BuildSystemDelegate? { get set }
+
+  /// Register the given file for build-system level change notifications, such as command
+  /// line flag changes, dependency changes, etc.
+  func registerChangeWatching(for: URL)
```

Comment:
Nitpick: I think `registerForChangeNotifications(_:)` might be a more grammatically correct name.

Reply:
Done